scvi/hub/hub_model.py

```diff
@@ -0,0 +1,284 @@
+import importlib
+import json
+import logging
+import os
+from dataclasses import asdict
+from pathlib import Path
+from typing import Optional, Type, Union
+
+import anndata
+import rich
+from anndata import AnnData
+from huggingface_hub import HfApi, ModelCard, create_repo, snapshot_download
+from rich.markdown import Markdown
+
+from scvi.data._download import _download
+from scvi.hub.hub_metadata import HubMetadata, HubModelCardHelper
+from scvi.model.base import BaseModelClass
+from scvi.model.base._utils import _load_saved_files
+
+from ._constants import _SCVI_HUB
+
+logger = logging.getLogger(__name__)
+
+
+class HubModel:
+    """
+    Provides functionality to interact with the scvi-hub backed by `huggingface <https://huggingface.co/models>`_.
+
+    Parameters
+    ----------
+    local_dir
+        Local directory where the data and pre-trained model reside.
+    metadata
+        Either an instance of :class:`~scvi.hub.HubMetadata` that contains the required metadata for this model,
+        or a path to a file on disk where this metadata can be read from.
+    model_card
+        The model card for this pre-trained model. Model card is a markdown file that describes the pre-trained
+        model/data and is displayed on huggingface. This can be either an instance of
+        :class:`~huggingface_hub.ModelCard` or an instance of :class:`~scvi.hub.HubModelCardHelper` that wraps
+        the model card or a path to a file on disk where the model card can be read from.
+    """
```

Comment:
examples in the docs are appreciated (in addition to the tutorial)

Reply:
I'll add a note to come back and add examples here once I write the tutorials. That way I know what's a good example would be..